[qa-1571] upgrade workbench libs and migrate to cats 3

andy7i opened the pull request:
QA-1571

This PR (which includes significant contributions from the CI mob) follows up on these PRs:
* https://github.com/broadinstitute/workbench-libs/pull/770
* https://github.com/broadinstitute/rawls/pull/1547
* https://github.com/broadinstitute/firecloud-develop/pull/2685
* https://github.com/broadinstitute/rawls/pull/1558
And will be completed (hopefully) by this PR:
* https://github.com/broadinstitute/rawls/pull/1556

Changes in this PR
Upgrade to cats3 for workbench-libs compatibility
Updates workbench libs 
Optimize imports across rawls (except for XMLs, which we do not want to alter)

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [ ] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell your tech lead (TL) that the PR exists if they want to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

andy7i committed:
upgrade sbt version as recommended by cats migration guide (https://typelevel.org/cats-effect/docs/migration-guide)

sbt 1.5.0 removes eviction warning, and replaces it with stricter eviction error
https://github.com/sbt/sbt/releases/tag/v1.5.0

```
[error] (rawlsCore / update) found version conflict(s) in library dependencies; some are suspected to be binary incompatible:
[error]
[error] 	* org.typelevel:cats-core_2.12:2.6.1 (semver-spec) is selected over {2.1.1, 1.6.0, 2.0.0, 2.0.0, 2.2.0}
[error] 	    +- org.broadinstitute.dsde:rawls-core_2.12:0.1        (depends on 2.6.1)
[error] 	    +- com.comcast:ip4s-core_2.12:3.0.3                   (depends on 2.6.1)
[error] 	    +- org.typelevel:cats-effect-kernel_2.12:3.2.9        (depends on 2.6.1)
[error] 	    +- org.http4s:http4s-core_2.12:0.21.23                (depends on 2.6.1)
[error] 	    +- org.typelevel:cats-mtl_2.12:1.2.1                  (depends on 2.6.1)
[error] 	    +- org.typelevel:log4cats-core_2.12:2.1.1             (depends on 2.6.1)
[error] 	    +- co.fs2:fs2-core_2.12:3.1.1                         (depends on 2.6.1)
[error] 	    +- io.circe:circe-core_2.12:0.14.1                    (depends on 2.6.1)
[error] 	    +- org.broadinstitute.dsde:workbench-util_2.12:0.1-1fd7bae7b-SNAP (depends on 2.6.1)
[error] 	    +- io.lemonlabs:scala-uri_2.12:3.0.0                  (depends on 2.2.0)
[error] 	    +- io.chrisdavenport:vault_2.12:2.0.0                 (depends on 2.0.0)
[error] 	    +- io.chrisdavenport:unique_2.12:2.0.0                (depends on 2.0.0)
[error] 	    +- io.chrisdavenport:log4cats-slf4j_2.12:0.3.0        (depends on 1.6.0)
[error] 	    +- com.github.sebruck:opencensus-scala-core_2.12:0.7.2 (depends on 2.1.1)
[error]
[error] 	* org.typelevel:log4cats-slf4j_2.12:2.1.1 (early-semver) is selected over 1.2.2
[error] 	    +- org.broadinstitute.dsde.workbench:workbench-util2_2.12:0.2-ae11b9f (depends on 2.1.1)
[error] 	    +- org.broadinstitute.dsde.workbench:workbench-google2_2.12:0.21-9d25534 (depends on 1.2.2)
[error]
[error] 	* co.fs2:fs2-core_2.12:3.1.1 (early-semver) is selected over {3.0.4, 2.5.6, 2.2.2, 3.0.4, 2.5.6, 2.2.2}
[error] 	    +- co.fs2:fs2-io_2.12:3.1.1                           (depends on 3.1.1)
[error] 	    +- org.http4s:jawn-fs2_2.12:1.0.0                     (depends on 2.2.2)
[error] 	    +- org.http4s:http4s-core_2.12:0.21.23                (depends on 2.5.6)
[error] 	    +- io.circe:circe-fs2_2.12:0.14.0                     (depends on 3.0.4)
[error]
[error]
[error] this can be overridden using libraryDependencySchemes or evictionErrorLevel
```

we need to upgrade many of these dependencies, and we may want to do that across the board here. (I've added this to our tech debt notes doc https://docs.google.com/document/d/1r0JWqBvmT9Y6JjT8sUrHTzNIE0Ed8gQt8iLhs0DmAbA/edit) The Leonardo repo keeps all of their dependencies up to date using a bot as well as using one version hash for all workbench-libs: https://github.com/DataBiosphere/leonardo/blob/develop/project/Dependencies.scala#L19. We may want to do something similar.

For now, there are 2 ways to opt out from this error (given by the sbt team: https://eed3si9n.com/sbt-1.5.0#:~:text=slash%20for%20details.-,Eviction%20error,-sbt%201.5.0%20removes) which is either to set all eviction errors back to INFO logs, or to specify which modules to ignore. I went with the latter option since it is more specific.

Side note: Intellij(?) made this really difficult. Changes that I made to build.sbt would not compile correctly until I successfully imported sbt 1.5.5 by running sbt build with only the sbt version change. Additionally, I would make changes to build.sbt (like commenting out the lines I added) and reload the sbt project expecting it to fail, but it would succeed instead. I needed to change other settings (I changed the scalav in Settings.scala) in order for it to pick up the changes in build.sbt.

andy7i committed:
upgrade sbt version as recommended by cats migration guide (https://typelevel.org/cats-effect/docs/migration-guide)

sbt 1.5.0 removes eviction warning, and replaces it with stricter eviction error
https://github.com/sbt/sbt/releases/tag/v1.5.0

```
[error] (rawlsCore / update) found version conflict(s) in library dependencies; some are suspected to be binary incompatible:
[error]
[error] 	* org.typelevel:cats-core_2.12:2.6.1 (semver-spec) is selected over {2.1.1, 1.6.0, 2.0.0, 2.0.0, 2.2.0}
[error] 	    +- org.broadinstitute.dsde:rawls-core_2.12:0.1        (depends on 2.6.1)
[error] 	    +- com.comcast:ip4s-core_2.12:3.0.3                   (depends on 2.6.1)
[error] 	    +- org.typelevel:cats-effect-kernel_2.12:3.2.9        (depends on 2.6.1)
[error] 	    +- org.http4s:http4s-core_2.12:0.21.23                (depends on 2.6.1)
[error] 	    +- org.typelevel:cats-mtl_2.12:1.2.1                  (depends on 2.6.1)
[error] 	    +- org.typelevel:log4cats-core_2.12:2.1.1             (depends on 2.6.1)
[error] 	    +- co.fs2:fs2-core_2.12:3.1.1                         (depends on 2.6.1)
[error] 	    +- io.circe:circe-core_2.12:0.14.1                    (depends on 2.6.1)
[error] 	    +- org.broadinstitute.dsde:workbench-util_2.12:0.1-1fd7bae7b-SNAP (depends on 2.6.1)
[error] 	    +- io.lemonlabs:scala-uri_2.12:3.0.0                  (depends on 2.2.0)
[error] 	    +- io.chrisdavenport:vault_2.12:2.0.0                 (depends on 2.0.0)
[error] 	    +- io.chrisdavenport:unique_2.12:2.0.0                (depends on 2.0.0)
[error] 	    +- io.chrisdavenport:log4cats-slf4j_2.12:0.3.0        (depends on 1.6.0)
[error] 	    +- com.github.sebruck:opencensus-scala-core_2.12:0.7.2 (depends on 2.1.1)
[error]
[error] 	* org.typelevel:log4cats-slf4j_2.12:2.1.1 (early-semver) is selected over 1.2.2
[error] 	    +- org.broadinstitute.dsde.workbench:workbench-util2_2.12:0.2-ae11b9f (depends on 2.1.1)
[error] 	    +- org.broadinstitute.dsde.workbench:workbench-google2_2.12:0.21-9d25534 (depends on 1.2.2)
[error]
[error] 	* co.fs2:fs2-core_2.12:3.1.1 (early-semver) is selected over {3.0.4, 2.5.6, 2.2.2, 3.0.4, 2.5.6, 2.2.2}
[error] 	    +- co.fs2:fs2-io_2.12:3.1.1                           (depends on 3.1.1)
[error] 	    +- org.http4s:jawn-fs2_2.12:1.0.0                     (depends on 2.2.2)
[error] 	    +- org.http4s:http4s-core_2.12:0.21.23                (depends on 2.5.6)
[error] 	    +- io.circe:circe-fs2_2.12:0.14.0                     (depends on 3.0.4)
[error]
[error]
[error] this can be overridden using libraryDependencySchemes or evictionErrorLevel
```

we need to upgrade many of these dependencies, and we may want to do that across the board here. (I've added this to our tech debt notes doc https://docs.google.com/document/d/1r0JWqBvmT9Y6JjT8sUrHTzNIE0Ed8gQt8iLhs0DmAbA/edit) The Leonardo repo keeps all of their dependencies up to date using a bot as well as using one version hash for all workbench-libs: https://github.com/DataBiosphere/leonardo/blob/develop/project/Dependencies.scala#L19. We may want to do something similar.

For now, there are 2 ways to opt out from this error (given by the sbt team: https://eed3si9n.com/sbt-1.5.0#:~:text=slash%20for%20details.-,Eviction%20error,-sbt%201.5.0%20removes) which is either to set all eviction errors back to INFO logs, or to specify which modules to ignore. I went with the latter option since it is more specific.

Side note: Intellij(?) made this really difficult. Changes that I made to build.sbt would not compile correctly until I successfully imported sbt 1.5.5 by running sbt build with only the sbt version change. Additionally, I would make changes to build.sbt (like commenting out the lines I added) and reload the sbt project expecting it to fail, but it would succeed instead. I needed to change other settings (I changed the scalav in Settings.scala) in order for it to pick up the changes in build.sbt.

andy7i committed:
upgrade sbt version as recommended by cats migration guide (https://typelevel.org/cats-effect/docs/migration-guide)

sbt 1.5.0 removes eviction warning, and replaces it with stricter eviction error
https://github.com/sbt/sbt/releases/tag/v1.5.0

```
[error] (rawlsCore / update) found version conflict(s) in library dependencies; some are suspected to be binary incompatible:
[error]
[error] 	* org.typelevel:cats-core_2.12:2.6.1 (semver-spec) is selected over {2.1.1, 1.6.0, 2.0.0, 2.0.0, 2.2.0}
[error] 	    +- org.broadinstitute.dsde:rawls-core_2.12:0.1        (depends on 2.6.1)
[error] 	    +- com.comcast:ip4s-core_2.12:3.0.3                   (depends on 2.6.1)
[error] 	    +- org.typelevel:cats-effect-kernel_2.12:3.2.9        (depends on 2.6.1)
[error] 	    +- org.http4s:http4s-core_2.12:0.21.23                (depends on 2.6.1)
[error] 	    +- org.typelevel:cats-mtl_2.12:1.2.1                  (depends on 2.6.1)
[error] 	    +- org.typelevel:log4cats-core_2.12:2.1.1             (depends on 2.6.1)
[error] 	    +- co.fs2:fs2-core_2.12:3.1.1                         (depends on 2.6.1)
[error] 	    +- io.circe:circe-core_2.12:0.14.1                    (depends on 2.6.1)
[error] 	    +- org.broadinstitute.dsde:workbench-util_2.12:0.1-1fd7bae7b-SNAP (depends on 2.6.1)
[error] 	    +- io.lemonlabs:scala-uri_2.12:3.0.0                  (depends on 2.2.0)
[error] 	    +- io.chrisdavenport:vault_2.12:2.0.0                 (depends on 2.0.0)
[error] 	    +- io.chrisdavenport:unique_2.12:2.0.0                (depends on 2.0.0)
[error] 	    +- io.chrisdavenport:log4cats-slf4j_2.12:0.3.0        (depends on 1.6.0)
[error] 	    +- com.github.sebruck:opencensus-scala-core_2.12:0.7.2 (depends on 2.1.1)
[error]
[error] 	* org.typelevel:log4cats-slf4j_2.12:2.1.1 (early-semver) is selected over 1.2.2
[error] 	    +- org.broadinstitute.dsde.workbench:workbench-util2_2.12:0.2-ae11b9f (depends on 2.1.1)
[error] 	    +- org.broadinstitute.dsde.workbench:workbench-google2_2.12:0.21-9d25534 (depends on 1.2.2)
[error]
[error] 	* co.fs2:fs2-core_2.12:3.1.1 (early-semver) is selected over {3.0.4, 2.5.6, 2.2.2, 3.0.4, 2.5.6, 2.2.2}
[error] 	    +- co.fs2:fs2-io_2.12:3.1.1                           (depends on 3.1.1)
[error] 	    +- org.http4s:jawn-fs2_2.12:1.0.0                     (depends on 2.2.2)
[error] 	    +- org.http4s:http4s-core_2.12:0.21.23                (depends on 2.5.6)
[error] 	    +- io.circe:circe-fs2_2.12:0.14.0                     (depends on 3.0.4)
[error]
[error]
[error] this can be overridden using libraryDependencySchemes or evictionErrorLevel
```

we need to upgrade many of these dependencies, and we may want to do that across the board here. (I've added this to our tech debt notes doc https://docs.google.com/document/d/1r0JWqBvmT9Y6JjT8sUrHTzNIE0Ed8gQt8iLhs0DmAbA/edit) The Leonardo repo keeps all of their dependencies up to date using a bot as well as using one version hash for all workbench-libs: https://github.com/DataBiosphere/leonardo/blob/develop/project/Dependencies.scala#L19. We may want to do something similar.

For now, there are 2 ways to opt out from this error (given by the sbt team: https://eed3si9n.com/sbt-1.5.0#:~:text=slash%20for%20details.-,Eviction%20error,-sbt%201.5.0%20removes) which is either to set all eviction errors back to INFO logs, or to specify which modules to ignore. I went with the latter option since it is more specific.

Side note: Intellij(?) made this really difficult. Changes that I made to build.sbt would not compile correctly until I successfully imported sbt 1.5.5 by running sbt build with only the sbt version change. Additionally, I would make changes to build.sbt (like commenting out the lines I added) and reload the sbt project expecting it to fail, but it would succeed instead. I needed to change other settings (I changed the scalav in Settings.scala) in order for it to pick up the changes in build.sbt.

andy7i committed:
update scala, sbt versions

we need to upgrade scalaV as well since there is no scala 2.12.11 with sbt 1.5.5)

andy7i committed:
update scala, sbt versions

we need to upgrade scalaV as well since there is no scala 2.12.11 with sbt 1.5.5)

andy7i committed:
one more

andy7i committed:
one more

andy7i committed:
try 2.12.14

andy7i committed:
try 2.12.14

andy7i committed:
sbt-scoverage upgrade

fixes errors saying:
```
java.lang.NoSuchMethodError: scala/tools/nsc/Global.reporter()Lscala/tools/nsc/reporters/Reporter
```

andy7i committed:
sbt-scoverage upgrade

fixes errors saying:
```
java.lang.NoSuchMethodError: scala/tools/nsc/Global.reporter()Lscala/tools/nsc/reporters/Reporter
```

andy7i committed:
Revert "try 2.12.14"

This reverts commit 44d0fce34ba6a6a68b8f2bc64e7c1b313e6cd468.

andy7i committed:
Revert "try 2.12.14"

This reverts commit 44d0fce34ba6a6a68b8f2bc64e7c1b313e6cd468.

andy7i committed:
// For the following errors:
```
    //[error] java.lang.RuntimeException: deduplicate: different file contents found in the following:
    //[error] /home/sbtuser/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.12.15/scala-library-2.12.15.jar:scala/annotation/nowarn$.class
    //[error] /home/sbtuser/.cache/coursier/v1/https/repo1.maven.org/maven2/org/http4s/http4s-core_2.12/0.21.23/http4s-core_2.12-0.21.23.jar:scala/annotation/nowarn$.class
    //[error] deduplicate: different file contents found in the following:
    //[error] /home/sbtuser/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.12.15/scala-library-2.12.15.jar:scala/annotation/nowarn.class
    //[error] /home/sbtuser/.cache/coursier/v1/https/repo1.maven.org/maven2/org/http4s/http4s-core_2.12/0.21.23/http4s-core_2.12-0.21.23.jar:scala/annotation/nowarn.class
```

andy7i committed:
// For the following errors:
```
    //[error] java.lang.RuntimeException: deduplicate: different file contents found in the following:
    //[error] /home/sbtuser/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.12.15/scala-library-2.12.15.jar:scala/annotation/nowarn$.class
    //[error] /home/sbtuser/.cache/coursier/v1/https/repo1.maven.org/maven2/org/http4s/http4s-core_2.12/0.21.23/http4s-core_2.12-0.21.23.jar:scala/annotation/nowarn$.class
    //[error] deduplicate: different file contents found in the following:
    //[error] /home/sbtuser/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.12.15/scala-library-2.12.15.jar:scala/annotation/nowarn.class
    //[error] /home/sbtuser/.cache/coursier/v1/https/repo1.maven.org/maven2/org/http4s/http4s-core_2.12/0.21.23/http4s-core_2.12-0.21.23.jar:scala/annotation/nowarn.class
```

andy7i committed:
Edmund ran sbt assembly without each line to see which ones were still used. This removes the lines that are no longer used.

andy7i committed:
Edmund ran sbt assembly without each line to see which ones were still used. This removes the lines that are no longer used.

andy7i committed:
need hash even for publishing rawls model

this handles the hash without needing to install git on the docker image (which, fun fact, does not have apt, yum, dpkg, or even sudo)

 ```
 14:42:55 [info] resolving key references (11031 settings) ...
 14:42:55 java.io.IOException: Cannot run program "git": error=2, No such file or directory
 14:42:55 	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1128)

```
from https://fc-jenkins.dsp-techops.broadinstitute.org/job/rawls-build/4805/console

andy7i committed:
need hash even for publishing rawls model

this handles the hash without needing to install git on the docker image (which, fun fact, does not have apt, yum, dpkg, or even sudo)

 ```
 14:42:55 [info] resolving key references (11031 settings) ...
 14:42:55 java.io.IOException: Cannot run program "git": error=2, No such file or directory
 14:42:55 	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1128)

```
from https://fc-jenkins.dsp-techops.broadinstitute.org/job/rawls-build/4805/console

andy7i committed:
get the git hash for the repo, not just the model

andy7i committed:
get the git hash for the repo, not just the model

andy7i committed:
PR feedback - rename GIT_HASH back to GIT_MODEL_HASH

andy7i committed:
PR feedback - rename GIT_HASH back to GIT_MODEL_HASH

andy7i committed:
use upgraded workbench-libs google to retry iam updates.

this would be all that's required if we had dependencies up to date.

andy7i committed:
Required more dependency merge conflict resolutions.

```

[error] (rawlsCore / update) found version conflict(s) in library dependencies; some are suspected to be binary incompatible:
[error]
[error] 	* org.typelevel:cats-core_2.12:2.6.1 (semver-spec) is selected over {2.1.1, 1.6.0, 2.0.0, 2.0.0, 2.2.0}
[error] 	    +- org.broadinstitute.dsde:rawls-core_2.12:0.1        (depends on 2.6.1)
[error] 	    +- com.comcast:ip4s-core_2.12:3.0.3                   (depends on 2.6.1)
[error] 	    +- org.typelevel:cats-effect-kernel_2.12:3.2.9        (depends on 2.6.1)
[error] 	    +- org.http4s:http4s-core_2.12:0.21.23                (depends on 2.6.1)
[error] 	    +- org.typelevel:cats-mtl_2.12:1.2.1                  (depends on 2.6.1)
[error] 	    +- org.typelevel:log4cats-core_2.12:2.1.1             (depends on 2.6.1)
[error] 	    +- co.fs2:fs2-core_2.12:3.1.1                         (depends on 2.6.1)
[error] 	    +- io.circe:circe-core_2.12:0.14.1                    (depends on 2.6.1)
[error] 	    +- org.broadinstitute.dsde:workbench-util_2.12:0.1-9f78b28b6-SNAP (depends on 2.6.1)
[error] 	    +- io.lemonlabs:scala-uri_2.12:3.0.0                  (depends on 2.2.0)
[error] 	    +- io.chrisdavenport:vault_2.12:2.0.0                 (depends on 2.0.0)
[error] 	    +- io.chrisdavenport:unique_2.12:2.0.0                (depends on 2.0.0)
[error] 	    +- io.chrisdavenport:log4cats-slf4j_2.12:0.3.0        (depends on 1.6.0)
[error] 	    +- com.github.sebruck:opencensus-scala-core_2.12:0.7.2 (depends on 2.1.1)
[error]
[error] 	* org.typelevel:log4cats-slf4j_2.12:2.1.1 (early-semver) is selected over 1.2.2
[error] 	    +- org.broadinstitute.dsde.workbench:workbench-util2_2.12:0.2-ae11b9f (depends on 2.1.1)
[error] 	    +- org.broadinstitute.dsde.workbench:workbench-google2_2.12:0.21-9d25534 (depends on 1.2.2)
[error]
[error] 	* co.fs2:fs2-core_2.12:3.1.1 (early-semver) is selected over {3.0.4, 2.5.6, 2.2.2, 3.0.4, 2.5.6, 2.2.2}
[error] 	    +- co.fs2:fs2-io_2.12:3.1.1                           (depends on 3.1.1)
[error] 	    +- org.http4s:jawn-fs2_2.12:1.0.0                     (depends on 2.2.2)
[error] 	    +- org.http4s:http4s-core_2.12:0.21.23                (depends on 2.5.6)
[error] 	    +- io.circe:circe-fs2_2.12:0.14.0                     (depends on 3.0.4)
[error]
[error]
[error] this can be overridden using libraryDependencySchemes or evictionErrorLevel
[error] (rawlsCore / ssExtractDependencies) found version conflict(s) in library dependencies; some are suspected to be binary incompatible:
[error]
[error] 	* org.typelevel:cats-core_2.12:2.6.1 (semver-spec) is selected over {2.1.1, 1.6.0, 2.0.0, 2.0.0, 2.2.0}
[error] 	    +- org.broadinstitute.dsde:rawls-core_2.12:0.1        (depends on 2.6.1)
[error] 	    +- com.comcast:ip4s-core_2.12:3.0.3                   (depends on 2.6.1)
[error] 	    +- org.typelevel:cats-effect-kernel_2.12:3.2.9        (depends on 2.6.1)
[error] 	    +- org.http4s:http4s-core_2.12:0.21.23                (depends on 2.6.1)
[error] 	    +- org.typelevel:cats-mtl_2.12:1.2.1                  (depends on 2.6.1)
[error] 	    +- org.typelevel:log4cats-core_2.12:2.1.1             (depends on 2.6.1)
[error] 	    +- co.fs2:fs2-core_2.12:3.1.1                         (depends on 2.6.1)
[error] 	    +- io.circe:circe-core_2.12:0.14.1                    (depends on 2.6.1)
[error] 	    +- org.broadinstitute.dsde:workbench-util_2.12:0.1-9f78b28b6-SNAP (depends on 2.6.1)
[error] 	    +- io.lemonlabs:scala-uri_2.12:3.0.0                  (depends on 2.2.0)
[error] 	    +- io.chrisdavenport:vault_2.12:2.0.0                 (depends on 2.0.0)
[error] 	    +- io.chrisdavenport:unique_2.12:2.0.0                (depends on 2.0.0)
[error] 	    +- io.chrisdavenport:log4cats-slf4j_2.12:0.3.0        (depends on 1.6.0)
[error] 	    +- com.github.sebruck:opencensus-scala-core_2.12:0.7.2 (depends on 2.1.1)
[error]
[error] 	* org.typelevel:log4cats-slf4j_2.12:2.1.1 (early-semver) is selected over 1.2.2
[error] 	    +- org.broadinstitute.dsde.workbench:workbench-util2_2.12:0.2-ae11b9f (depends on 2.1.1)
[error] 	    +- org.broadinstitute.dsde.workbench:workbench-google2_2.12:0.21-9d25534 (depends on 1.2.2)
[error]
[error] 	* co.fs2:fs2-core_2.12:3.1.1 (early-semver) is selected over {3.0.4, 2.5.6, 2.2.2, 3.0.4, 2.5.6, 2.2.2}
[error] 	    +- co.fs2:fs2-io_2.12:3.1.1                           (depends on 3.1.1)
[error] 	    +- org.http4s:jawn-fs2_2.12:1.0.0                     (depends on 2.2.2)
[error] 	    +- org.http4s:http4s-core_2.12:0.21.23                (depends on 2.5.6)
[error] 	    +- io.circe:circe-fs2_2.12:0.14.0                     (depends on 3.0.4)
[error]
[error]
[error] this can be overridden using libraryDependencySchemes or evictionErrorLevel
[error] (ssExtractDependencies) found version conflict(s) in library dependencies; some are suspected to be binary incompatible:
[error]
[error] 	* org.typelevel:cats-core_2.12:2.6.1 (semver-spec) is selected over {2.1.1, 1.6.0, 2.0.0, 2.0.0, 2.2.0}
[error] 	    +- org.broadinstitute.dsde:rawls-core_2.12:0.1        (depends on 2.6.1)
[error] 	    +- com.comcast:ip4s-core_2.12:3.0.3                   (depends on 2.6.1)
[error] 	    +- org.typelevel:cats-effect-kernel_2.12:3.2.9        (depends on 2.6.1)
[error] 	    +- org.http4s:http4s-core_2.12:0.21.23                (depends on 2.6.1)
[error] 	    +- org.typelevel:cats-mtl_2.12:1.2.1                  (depends on 2.6.1)
[error] 	    +- org.typelevel:log4cats-core_2.12:2.1.1             (depends on 2.6.1)
[error] 	    +- co.fs2:fs2-core_2.12:3.1.1                         (depends on 2.6.1)
[error] 	    +- io.circe:circe-core_2.12:0.14.1                    (depends on 2.6.1)
[error] 	    +- org.broadinstitute.dsde:workbench-util_2.12:0.1-9f78b28b6-SNAP (depends on 2.6.1)
[error] 	    +- io.lemonlabs:scala-uri_2.12:3.0.0                  (depends on 2.2.0)
[error] 	    +- io.chrisdavenport:vault_2.12:2.0.0                 (depends on 2.0.0)
[error] 	    +- io.chrisdavenport:unique_2.12:2.0.0                (depends on 2.0.0)
[error] 	    +- io.chrisdavenport:log4cats-slf4j_2.12:0.3.0        (depends on 1.6.0)
[error] 	    +- com.github.sebruck:opencensus-scala-core_2.12:0.7.2 (depends on 2.1.1)
[error]
[error] 	* org.typelevel:log4cats-slf4j_2.12:2.1.1 (early-semver) is selected over 1.2.2
[error] 	    +- org.broadinstitute.dsde.workbench:workbench-util2_2.12:0.2-ae11b9f (depends on 2.1.1)
[error] 	    +- org.broadinstitute.dsde.workbench:workbench-google2_2.12:0.21-9d25534 (depends on 1.2.2)
[error]
[error] 	* co.fs2:fs2-core_2.12:3.1.1 (early-semver) is selected over {3.0.4, 2.5.6, 2.2.2, 3.0.4, 2.5.6, 2.2.2}
[error] 	    +- co.fs2:fs2-io_2.12:3.1.1                           (depends on 3.1.1)
[error] 	    +- org.http4s:jawn-fs2_2.12:1.0.0                     (depends on 2.2.2)
[error] 	    +- org.http4s:http4s-core_2.12:0.21.23                (depends on 2.5.6)
[error] 	    +- io.circe:circe-fs2_2.12:0.14.0                     (depends on 3.0.4)
[error]
[error]
```

andy7i committed:
Required more dependency merge conflict resolutions.

```

[error] (rawlsCore / update) found version conflict(s) in library dependencies; some are suspected to be binary incompatible:
[error]
[error] 	* org.typelevel:cats-core_2.12:2.6.1 (semver-spec) is selected over {2.1.1, 1.6.0, 2.0.0, 2.0.0, 2.2.0}
[error] 	    +- org.broadinstitute.dsde:rawls-core_2.12:0.1        (depends on 2.6.1)
[error] 	    +- com.comcast:ip4s-core_2.12:3.0.3                   (depends on 2.6.1)
[error] 	    +- org.typelevel:cats-effect-kernel_2.12:3.2.9        (depends on 2.6.1)
[error] 	    +- org.http4s:http4s-core_2.12:0.21.23                (depends on 2.6.1)
[error] 	    +- org.typelevel:cats-mtl_2.12:1.2.1                  (depends on 2.6.1)
[error] 	    +- org.typelevel:log4cats-core_2.12:2.1.1             (depends on 2.6.1)
[error] 	    +- co.fs2:fs2-core_2.12:3.1.1                         (depends on 2.6.1)
[error] 	    +- io.circe:circe-core_2.12:0.14.1                    (depends on 2.6.1)
[error] 	    +- org.broadinstitute.dsde:workbench-util_2.12:0.1-9f78b28b6-SNAP (depends on 2.6.1)
[error] 	    +- io.lemonlabs:scala-uri_2.12:3.0.0                  (depends on 2.2.0)
[error] 	    +- io.chrisdavenport:vault_2.12:2.0.0                 (depends on 2.0.0)
[error] 	    +- io.chrisdavenport:unique_2.12:2.0.0                (depends on 2.0.0)
[error] 	    +- io.chrisdavenport:log4cats-slf4j_2.12:0.3.0        (depends on 1.6.0)
[error] 	    +- com.github.sebruck:opencensus-scala-core_2.12:0.7.2 (depends on 2.1.1)
[error]
[error] 	* org.typelevel:log4cats-slf4j_2.12:2.1.1 (early-semver) is selected over 1.2.2
[error] 	    +- org.broadinstitute.dsde.workbench:workbench-util2_2.12:0.2-ae11b9f (depends on 2.1.1)
[error] 	    +- org.broadinstitute.dsde.workbench:workbench-google2_2.12:0.21-9d25534 (depends on 1.2.2)
[error]
[error] 	* co.fs2:fs2-core_2.12:3.1.1 (early-semver) is selected over {3.0.4, 2.5.6, 2.2.2, 3.0.4, 2.5.6, 2.2.2}
[error] 	    +- co.fs2:fs2-io_2.12:3.1.1                           (depends on 3.1.1)
[error] 	    +- org.http4s:jawn-fs2_2.12:1.0.0                     (depends on 2.2.2)
[error] 	    +- org.http4s:http4s-core_2.12:0.21.23                (depends on 2.5.6)
[error] 	    +- io.circe:circe-fs2_2.12:0.14.0                     (depends on 3.0.4)
[error]
[error]
[error] this can be overridden using libraryDependencySchemes or evictionErrorLevel
[error] (rawlsCore / ssExtractDependencies) found version conflict(s) in library dependencies; some are suspected to be binary incompatible:
[error]
[error] 	* org.typelevel:cats-core_2.12:2.6.1 (semver-spec) is selected over {2.1.1, 1.6.0, 2.0.0, 2.0.0, 2.2.0}
[error] 	    +- org.broadinstitute.dsde:rawls-core_2.12:0.1        (depends on 2.6.1)
[error] 	    +- com.comcast:ip4s-core_2.12:3.0.3                   (depends on 2.6.1)
[error] 	    +- org.typelevel:cats-effect-kernel_2.12:3.2.9        (depends on 2.6.1)
[error] 	    +- org.http4s:http4s-core_2.12:0.21.23                (depends on 2.6.1)
[error] 	    +- org.typelevel:cats-mtl_2.12:1.2.1                  (depends on 2.6.1)
[error] 	    +- org.typelevel:log4cats-core_2.12:2.1.1             (depends on 2.6.1)
[error] 	    +- co.fs2:fs2-core_2.12:3.1.1                         (depends on 2.6.1)
[error] 	    +- io.circe:circe-core_2.12:0.14.1                    (depends on 2.6.1)
[error] 	    +- org.broadinstitute.dsde:workbench-util_2.12:0.1-9f78b28b6-SNAP (depends on 2.6.1)
[error] 	    +- io.lemonlabs:scala-uri_2.12:3.0.0                  (depends on 2.2.0)
[error] 	    +- io.chrisdavenport:vault_2.12:2.0.0                 (depends on 2.0.0)
[error] 	    +- io.chrisdavenport:unique_2.12:2.0.0                (depends on 2.0.0)
[error] 	    +- io.chrisdavenport:log4cats-slf4j_2.12:0.3.0        (depends on 1.6.0)
[error] 	    +- com.github.sebruck:opencensus-scala-core_2.12:0.7.2 (depends on 2.1.1)
[error]
[error] 	* org.typelevel:log4cats-slf4j_2.12:2.1.1 (early-semver) is selected over 1.2.2
[error] 	    +- org.broadinstitute.dsde.workbench:workbench-util2_2.12:0.2-ae11b9f (depends on 2.1.1)
[error] 	    +- org.broadinstitute.dsde.workbench:workbench-google2_2.12:0.21-9d25534 (depends on 1.2.2)
[error]
[error] 	* co.fs2:fs2-core_2.12:3.1.1 (early-semver) is selected over {3.0.4, 2.5.6, 2.2.2, 3.0.4, 2.5.6, 2.2.2}
[error] 	    +- co.fs2:fs2-io_2.12:3.1.1                           (depends on 3.1.1)
[error] 	    +- org.http4s:jawn-fs2_2.12:1.0.0                     (depends on 2.2.2)
[error] 	    +- org.http4s:http4s-core_2.12:0.21.23                (depends on 2.5.6)
[error] 	    +- io.circe:circe-fs2_2.12:0.14.0                     (depends on 3.0.4)
[error]
[error]
[error] this can be overridden using libraryDependencySchemes or evictionErrorLevel
[error] (ssExtractDependencies) found version conflict(s) in library dependencies; some are suspected to be binary incompatible:
[error]
[error] 	* org.typelevel:cats-core_2.12:2.6.1 (semver-spec) is selected over {2.1.1, 1.6.0, 2.0.0, 2.0.0, 2.2.0}
[error] 	    +- org.broadinstitute.dsde:rawls-core_2.12:0.1        (depends on 2.6.1)
[error] 	    +- com.comcast:ip4s-core_2.12:3.0.3                   (depends on 2.6.1)
[error] 	    +- org.typelevel:cats-effect-kernel_2.12:3.2.9        (depends on 2.6.1)
[error] 	    +- org.http4s:http4s-core_2.12:0.21.23                (depends on 2.6.1)
[error] 	    +- org.typelevel:cats-mtl_2.12:1.2.1                  (depends on 2.6.1)
[error] 	    +- org.typelevel:log4cats-core_2.12:2.1.1             (depends on 2.6.1)
[error] 	    +- co.fs2:fs2-core_2.12:3.1.1                         (depends on 2.6.1)
[error] 	    +- io.circe:circe-core_2.12:0.14.1                    (depends on 2.6.1)
[error] 	    +- org.broadinstitute.dsde:workbench-util_2.12:0.1-9f78b28b6-SNAP (depends on 2.6.1)
[error] 	    +- io.lemonlabs:scala-uri_2.12:3.0.0                  (depends on 2.2.0)
[error] 	    +- io.chrisdavenport:vault_2.12:2.0.0                 (depends on 2.0.0)
[error] 	    +- io.chrisdavenport:unique_2.12:2.0.0                (depends on 2.0.0)
[error] 	    +- io.chrisdavenport:log4cats-slf4j_2.12:0.3.0        (depends on 1.6.0)
[error] 	    +- com.github.sebruck:opencensus-scala-core_2.12:0.7.2 (depends on 2.1.1)
[error]
[error] 	* org.typelevel:log4cats-slf4j_2.12:2.1.1 (early-semver) is selected over 1.2.2
[error] 	    +- org.broadinstitute.dsde.workbench:workbench-util2_2.12:0.2-ae11b9f (depends on 2.1.1)
[error] 	    +- org.broadinstitute.dsde.workbench:workbench-google2_2.12:0.21-9d25534 (depends on 1.2.2)
[error]
[error] 	* co.fs2:fs2-core_2.12:3.1.1 (early-semver) is selected over {3.0.4, 2.5.6, 2.2.2, 3.0.4, 2.5.6, 2.2.2}
[error] 	    +- co.fs2:fs2-io_2.12:3.1.1                           (depends on 3.1.1)
[error] 	    +- org.http4s:jawn-fs2_2.12:1.0.0                     (depends on 2.2.2)
[error] 	    +- org.http4s:http4s-core_2.12:0.21.23                (depends on 2.5.6)
[error] 	    +- io.circe:circe-fs2_2.12:0.14.0                     (depends on 3.0.4)
[error]
[error]
```

andy7i committed:
upgrade and prepare scalafix for cats effects 3 migration

andy7i committed:
cats effects 3 migration:
```
sbt ";scalafixEnable ;scalafixAll github:typelevel/cats-effect/v3_0_0?sha=v3.0.0"
```

andy7i committed:
cats effects migration doc says we should upgrade all the things so that all versions of cats effects are updated. Thankfully, this seems to have eliminated all dependency issues so we can remove some of these merge strategies as well.

andy7i committed:
cats effects migration doc says we should upgrade all the things so that all versions of cats effects are updated. Thankfully, this seems to have eliminated all dependency issues so we can remove some of these merge strategies as well.

andy7i committed:
dummy commit

andy7i committed:
wip - mob fixing migration errors

ehigham committed:
run "Optimize Imports" on whole project

ehigham committed:
it builds...

ehigham committed:
re-optimise

marctalbott commented on automation/project/Dependencies.scala:
```diff
@@ -8,7 +8,7 @@ object Dependencies {
 
   val serviceTestV = "0.20-21408a9"
-  val workbenchGoogleV = "0.21-21408a9"
+  val workbenchGoogleV = "0.21-ae11b9f"
```
Can we continue to use the same hash across libs here as well?

Qi77Qi replied:
btw, with this, if there're `exclude` rules for any workbench libs, those rules can all be removed

Qi77Qi replied:
like [those](https://github.com/broadinstitute/rawls/blob/develop/project/Dependencies.scala#L21-L22)

andy7i committed:
Merge branch 'develop' into aj-sr--upgrade-sbt-version--qa-1571

andy7i committed:
cherry-pick reverting changes made to liquibase XMLs. "optimize imports" changed a lot of the liquibase xmls, which is not ideal because liquibase takes the hash of the xml and reruns it if the hash has changed. (the 95 .xml files were changed in this commit https://github.com/broadinstitute/rawls/pull/1554/commits/c3e37ecc258a751cf8dd756ebbc0b173e21c3d4d)

andy7i committed:
intellij still edited many of the xmls, so this time I ran `git checkout develop -- core/src/main/resources/org/broadinstitute/dsde/rawls/liquibase/changesets` to get the original versions

marctalbott commented on core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/SamDAO.scala:
```diff
@@ -82,5 +82,5 @@ object SamDAO {
     com.google.api.services.oauth2.Oauth2Scopes.USERINFO_PROFILE
   )
-  val bigQueryReadOnlyScope = com.google.api.services.bigquery.BigqueryScopes.BIGQUERY_READONLY
+  val bigQueryReadOnlyScope = com.google.api.services.bigquery.BigqueryScopes.DEVSTORAGE_READ_ONLY
```
Can we get someone from AppServices to confirm that this is the correct scope to use?

davidangb replied:
I can't speak as to why `BigqueryScopes.BIGQUERY_READONLY` is no longer available in the BQ client lib.

Following the constants through to their actual String values, I can see that:
```
  public static final String BIGQUERY_READONLY = "https://www.googleapis.com/auth/bigquery.readonly";
  public static final String DEVSTORAGE_READ_ONLY = "https://www.googleapis.com/auth/devstorage.read_only";
```

So this is definitely a change in functionality, per https://developers.google.com/identity/protocols/oauth2/scopes#bigquery, and I would expect this to break some features.

Trying to figure out why this might have been removed …

davidangb replied:
well, regardless of why this scope was removed from the client library, it's no longer needed. The val `bigQueryReadOnlyScope` is not used anywhere. As of #1477 it's an orphan, so I say just delete it and then you don't have to worry about what scope to use here. (We could do that in a separate PR and then rebase this one, if it keeps this PR cleaner)

andy7i replied:
nice find! I think having a new PR would be cleaner. Do you want to create the PR? Otherwise I can create it on Monday.

davidangb replied:
I just created a PR at #1558.

andy7i replied:
merged it in! thanks @davidangb

marctalbott commented on core/src/main/scala/org/broadinstitute/dsde/rawls/status/StatusService.scala:
```diff
@@ -18,7 +18,7 @@ import scala.language.postfixOps
   * Created by rtitle on 5/20/17.
   */
 object StatusService {
-  def constructor(healthMonitor: ActorRef)()(implicit executionContext: ExecutionContext): StatusService = {
```
Was removing this `()` intentional? I wonder if this is what caused our headache with the `_` yesterday

ehigham replied:
It is! Good spot.

andy7i replied:
reverted these changes!

marctalbott commented on core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala:
```diff
@@ -2045,7 +2043,7 @@ class WorkspaceService(protected val userInfo: UserInfo,
     // This is done sequentially intentionally in order to avoid conflict exceptions as a result of concurrent IAM updates.
     policyGroupsToRoles.toList.foldLeft(Future(true)){case (result, (email, roles)) => {
-      result.flatMap(_ => googleIamDao.addIamRoles(GoogleProject(googleProject.value), email, MemberType.Group, roles))
+      result.flatMap(_ => googleIamDao.addIamRoles(GoogleProject(googleProject.value), email, MemberType.Group, roles, retryIfGroupDoesNotExist = true))
```
Is this change meant to be in this PR? Slight preference to keep this PR limited to changes related to import optimization and the cats upgrade and add this change in a separate PR

andy7i replied:
I'd be ok putting it into a separate PR; fyi this is **the** fix I wanted to add for QA-1571. (everything else was necessary yak shaving)

marctalbott replied:
Up to you, but at the moment this change is kind of buried amongst the other changes. Since the cats upgrade is sort of pre-work for this change, it makes sense to me to have the cats upgrade go in first, and then follow up with a separate PR

andy7i replied:
created https://github.com/broadinstitute/rawls/pull/1556

marctalbott commented on project/Dependencies.scala:
```diff
@@ -74,13 +73,13 @@ object Dependencies {
   val commonsJEXL: ModuleID =     "org.apache.commons"            % "commons-jexl"          % "2.1.1"
   val httpClient: ModuleID =      "org.apache.httpcomponents"     % "httpclient"            % "4.5.3"  // upgrading a transitive dependency to avoid security warnings
-  val cats: ModuleID =            "org.typelevel"                 %% "cats-core"                 % "2.2.0"
+  val cats: ModuleID =            "org.typelevel"                 %% "cats-core"                 % "2.6.1"
```
Should we upgrade our version of cats to 3?

andy7i replied:
2.6.1 is the latest version of **cats-core**
https://mvnrepository.com/artifact/org.typelevel/cats-core

andy7i committed:
fixes test error

in general, better to return value (e.g. Option) than throwing an exception in order to properly handle / think through unexpected outcomes.

andy7i committed:
update mockito for glory

andy7i committed:
PR feedback -- the root cause of the mysterious error we saw before. reverts the changes.

andy7i committed:
PR feedback -- use wb-libs hash in automation as well

andy7i committed:
Revert "PR feedback - rename GIT_HASH back to GIT_MODEL_HASH"

This reverts commit 9f78b28b68b2d7a96d65da0a4f1e716c5d15e7a8.

andy7i committed:
change back to def. ; this fixes 403s from jfrog when there are no changes to model but we still need to publish new changes.
not 100% sure how this worked before but without spending time to investigate my guess is that it was actually using the repo's git hash (and NOT the model git hash).

example:
first PR:
```

11:58:01 [info] 	published rawls-model_2.12 to https://broadinstitute.jfrog.io/broadinstitute/libs-release-local;build.timestamp=1634918248/org/broadinstitute/dsde/rawls-model_2.12/0.1-c3e37ec-SNAP/rawls-model_2.12-0.1-c3e37ec-SNAP.pom
11:58:01 [info] 	published rawls-model_2.12 to https://broadinstitute.jfrog.io/broadinstitute/libs-release-local;build.timestamp=1634918248/org/broadinstitute/dsde/rawls-model_2.12/0.1-c3e37ec-SNAP/rawls-model_2.12-0.1-c3e37ec-SNAP.jar
11:58:02 [info] 	published rawls-model_2.12 to https://broadinstitute.jfrog.io/broadinstitute/libs-release-local;build.timestamp=1634918248/org/broadinstitute/dsde/rawls-model_2.12/0.1-c3e37ec-SNAP/rawls-model_2.12-0.1-c3e37ec-SNAP-sources.jar
11:58:02 [info] 	published rawls-model_2.12 to https://broadinstitute.jfrog.io/broadinstitute/libs-release-local;build.timestamp=1634918248/org/broadinstitute/dsde/rawls-model_2.12/0.1-c3e37ec-SNAP/rawls-model_2.12-0.1-c3e37ec-SNAP-javadoc.jar
```
from https://fc-jenkins.dsp-techops.broadinstitute.org/job/rawls-build/4850/console

subsequent commits all had this error:
```
12:00:54 [error] java.io.IOException: Server returned HTTP response code: 403 for URL: https://broadinstitute.jfrog.io/broadinstitute/libs-release-local;build.timestamp=1634918418/org/broadinstitute/dsde/rawls-model_2.12/0.1-c3e37ec-SNAP/rawls-model_2.12-0.1-c3e37ec-SNAP.pom
12:00:54 [error] 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)

```
https://fc-jenkins.dsp-techops.broadinstitute.org/job/rawls-build/4851/console

andy7i committed:
Merge branch 'aj-sr--upgrade-sbt-version--qa-1571' into aj-upgrade-migrate-cats-3--qa-1571

andy7i committed:
Revert "dummy commit"

This reverts commit 03fce34843b36e1275543eb865904871618e9fb0.

andy7i committed:
Merge branch 'aj-sr--upgrade-sbt-version--qa-1571' into aj-upgrade-migrate-cats-3--qa-1571

andy7i committed:
PR feedback -- separate out into another PR

andy7i committed:
Merge branch 'develop' into aj-upgrade-migrate-cats-3--qa-1571

andy7i commented on core/src/main/scala/org/broadinstitute/dsde/rawls/Boot.scala:
```diff
@@ -521,7 +520,7 @@ object Boot extends IOApp with LazyLogging {
   }
 
-  def initAppDependencies[F[_]: ConcurrentEffect: Timer: Logger: ContextShift](config: Config, appName: String, metricsPrefix: String)(implicit executionContext: ExecutionContext, system: ActorSystem): cats.effect.Resource[F, AppDependencies[F]] = {
+  def initAppDependencies[F[_]: Temporal: Logger: Async](config: Config, appName: String, metricsPrefix: String)(implicit executionContext: ExecutionContext, system: ActorSystem): cats.effect.Resource[F, AppDependencies[F]] = {
```
@Qi77Qi do the changes in this file look reasonable for the cats effects 3 migration?

Qi77Qi replied:
you don't need `Temporal` if you have `Async` already...`Async` is basically the most powerful one...so if you have that, unlikely you'll need anything else

andy7i commented on core/src/main/scala/org/broadinstitute/dsde/rawls/entities/datarepo/DataRepoBigQuerySupport.scala:
```diff
@@ -1,17 +1,16 @@
 import bio.terra.datarepo.model.{RelationshipModel, SnapshotModel, TableModel}
 import com.google.cloud.bigquery.Field.Mode
-import com.google.cloud.bigquery._
+import com.google.cloud.bigquery.{Option => _, _}
```
this notation is used to exclude the bigquery `Option` which was causing problems

andy7i committed:
self-review -- remove unnecessary dependency handling

andy7i committed:
Merge branch 'develop' into aj-upgrade-migrate-cats-3--qa-1571

andy7i commented on build.sbt:
```diff
@@ -1,9 +1,6 @@
 import Settings._
 import Testing._
 
-ThisBuild / libraryDependencySchemes += "org.typelevel" %% "cats-effect" % "always"
-ThisBuild / libraryDependencySchemes += "co.fs2" %% "fs2-io" % "always"
-
```
no longer needed, being fully migrated to CE3

andy7i commented on core/src/test/scala/org/broadinstitute/dsde/rawls/deltalayer/DeltaLayerSpec.scala:
```diff
@@ -397,7 +395,7 @@ class DeltaLayerSpec extends AsyncFreeSpec with TestDriverComponent with Private
         // mock GoogleBigQueryService that throws error on deleteDataset
         val throwingBQService = mock[GoogleBigQueryService[IO]]
-        when(throwingBQService.deleteDataset(any())).thenThrow(new BigQueryException(444, "BigQuery errors via deleteDataset should bubble up"))
+        when(throwingBQService.deleteDataset(any())).thenReturn(IO.raiseError(new BigQueryException(444, "BigQuery errors via deleteDataset should bubble up")))
```
fixes test error

andy7i commented on project/Dependencies.scala:
```diff
@@ -1,13 +1,12 @@
 object Dependencies {
-  val akkaV = "2.6.8"
-  val akkaHttpV = "10.2.0"
+  val akkaV = "2.6.17"
+  val akkaHttpV = "10.2.6"
```
upgraded a few other dependencies while in here.

andy7i commented on automation/project/Dependencies.scala:
```diff
@@ -7,11 +7,12 @@ object Dependencies {
-  val workbenchGoogle2V = "0.22-21408a9"
-  val workbenchModelV  = "0.14-21408a9"
-  val workbenchMetricsV  = "0.5-21408a9"
+  val workbenchLibsHash = "ae11b9f"
```
updated both Dependencies files to use a single workbench-libs hash.

andy7i reviewed (commented):
99.99% of this PR is Optimized Imports. I commented on the files that contain more interesting changes.

andy7i commented:
tests are passing, with the exception of codecov. Since this PR does not add any new logic, I'm inclined to ignore the codecov failure. Please add a thumb if that is ok, and I will ignore codecov for this PR.

andy7i commented:
jenkins retest -- seemingly unrelated error in orch, with a missing file during config generation.
```
ls: /working/configs/firecloud-orchestration/extra/*.ctmpl: No such file or directory
./run-api-tests.sh: line 112: target/firecloud-account.json: Permission denied
write /dev/stdout: broken pipe
testOnly -- -l ProdTest -l NotebooksCanaryTest -l ExcludeInFiab
java.lang.NoClassDefFoundError: Could not initialize class sbt.Keys$
	at sbt.xMain$.$anonfun$run$8(Main.scala:99)
	at scala.util.DynamicVariable.withValue(DynamicVariable.scala:62)
	at scala.Console$.withIn(Console.scala:230)
	at sbt.internal.util.Terminal$.withIn(Terminal.scala:560)
	at sbt.internal.util.Terminal$.$anonfun$withStreams$1(Terminal.scala:350)
	at scala.util.DynamicVariable.withValue(DynamicVariable.scala:62)
	at scala.Console$.withOut(Console.scala:167)
	at sbt.internal.util.Terminal$.$anonfun$withOut$2(Terminal.scala:550)
	at scala.util.DynamicVariable.withValue(DynamicVariable.scala:62)
	at scala.Console$.withErr(Console.scala:196)
	at sbt.internal.util.Terminal$.withOut(Terminal.scala:550)
	at sbt.internal.util.Terminal$.withStreams(Terminal.scala:350)
	at sbt.xMain$.run(Main.scala:86)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
```

Qi77Qi commented:
🎉 :shipit:   ...sam is the next! haha

andy7i commented:
jenkins retest -- seeing the same orch config issue ; not seeing any recent changes in fc-dev or orch so rerunning.

Qi77Qi commented:
@andy7i  can you try running the the same sbt cmd locally and see if it passes....I don't rememer the details, but is there some orch lib that rawls depends on? if yes, is that lib on cats-effect 3 (if that lib depends on cats-effect)?

andy7i commented:
jenkins retest -- orch passed here: https://fc-jenkins.dsp-techops.broadinstitute.org/job/swatomation-pipeline/56024/
but github did not track this retest. (I kicked it off before the prior test had finished.)

andy7i committed:
PR feedback -- "you don't need Temporal if you have Async already...Async is basically the most powerful one...so if you have that, unlikely you'll need anything else"

andy7i committed:
PR feedback -- since workbench libs are synced, we can exclude these exclusion rules
